README.md

```diff
@@ -1,31 +1,114 @@
-# PyRS990
+![PyRS990 Header](pyrs990_header.png)
 
 It's a pun. Get it?
 
-A Python application that can grab all sorts of IRS Form 990 data on non-profit organizations and
-put it into a format that can be consumed easily.
+A Python application and library that can grab all sorts of IRS Form 990
+data on non-profit organizations and put it into a format that can be
+consumed easily by other applications.
 
-## Running
+## Up and Running
+
+The instructions below should allow you to get the software working
+for your purpose (user or developer). If you run into trouble please,
+please let us know so that we can update the instructions (or fix the
+bug you ran into).
+
+### User
 
 For now you need to clone the repo to use it. Eventually we'll package it.
 
   1. Make sure you have Python 3.8 available
   1. Install `pipenv` - `pip install --user pipenv`
   1. Clone the whole repo, `cd` into the `pyrs990` directory
   1. Install dependencies - `pipenv sync`
-  1. Run the example: `pipenv run python -m pyrs990`
-  1. Run it again to use the cache (notice the speedup)
+  1. Run it, some very simple examples are below:
+      1. `pipenv run python -m pyrs990 --zip 59801 --use-disk-cache`
+      1. ...more examples coming soon
+  1. Run the commands again, notice the cache speedup
   1. The cache is set to `./.pyrs990-cache/`
 
-## Development
+### Developer
 
   1. Make sure you have Python 3.8 available
   1. Install `pipenv` - `pip install --user pipenv`
   1. Clone the whole repo, `cd` into the `pyrs990` directory
-  1. Install dependencies - `pipenv install --dev`
+  1. Install dependencies - `pipenv sync --dev`
   1. Make your changes
   1. If you need to add dependencies:
     1. `pipenv install coolpkg`
     1. `pipenv lock` (updates the lock file)
   1. Make a pull request
 
+## About the Data
+
+Right now we pull data that originated with the IRS (hence the silly name)
+but we get it from a couple sources and information about what is actually
+available is a little spread out as well.
+
+### Structure
+
+There are two indices used to narrow down the list of filing documents
+that must be downloaded a satisfy a given query. The first is an
+annual index (we refer to it as "Annual" or "Annual Index" in the
+code). This index contains all filings processed by the IRS for a
+given calendar year.
+
+Note that this does not necessarily have anything
+to do with the filing year. An organization might, for example, file
+its 2016 990 in either 2017 or 2018 (or even later). There is a field,
+described below, called `tax_period` that reflects the filing period.
+In the future, we intend to further abstract this is that it is
```

Comment:
#nit we intend to further abstract this is that -> we intend to further abstract this so that